frontend/src/features/comparisonSeries/ComparisonSeries.tsx

```diff
@@ -50,32 +74,72 @@ const ComparisonSeries = ({
-  // same comparison to a user
+  // an array of already made comparisons, allowing to not suggest two times the
+  // same comparison to a user, formatted like this ['uidA/uidB', 'uidA/uidC']
   const [comparisonsMade, setComparisonsMade] = useState<Array<string>>([]);
+  // a string representing the URL parameters of the first comparison that may be suggested
+  const [firstComparisonParams, setFirstComparisonParams] = useState('');
 
   const searchParams = new URLSearchParams(location.search);
   const uidA: string = searchParams.get(UID_PARAMS.vidA) || '';
   const uidB: string = searchParams.get(UID_PARAMS.vidB) || '';
 
   /**
-   * Build the list of `alternatives`.
+   * Retrieve the user's comparisons to avoid suggesting couples of entities
+   * that have already been compared.
+   *
+   * Also build the list of `alternatives`. After each comparison, an entity
+   * from this list can be selected to replace one of the two compared
+   * entities.
    *
-   * After each comparison, an entity from this list can be selected to
-   * replace one of the two compared entities.
+   * If the component has been mounted with `generateInitial` = true, build the
+   * URL parameters that will be used to suggest the first comparison of the
+   * series.
    */
   useEffect(() => {
-    async function getAlternativesAsync() {
-      if (getAlternatives) {
-        const alts = await getAlternatives();
-        if (alts.length > 0) {
-          setAlternatives(alts);
-        }
+    async function getAlternativesAsync(getAlts: () => Promise<Array<Entity>>) {
+      const alts = await getAlts();
+      if (alts.length > 0) {
+        setAlternatives(alts);
+        return alts;
       }
+      return [];
     }
 
-    getAlternativesAsync();
-  }, [getAlternatives]);
+    async function getUserComparisonsAsync(pName: string) {
+      const comparisons = await getUserComparisons(pName);
+      const formattedComparisons = comparisons.map(
+        (c) => c.entity_a.uid + '/' + c.entity_b.uid
+      );
+
+      setComparisonsMade(formattedComparisons);
+      return formattedComparisons;
+    }
+
+    if (length >= MIN_LENGTH) {
+      getUserComparisonsAsync(pollName).then((comparisons) => {
+        if (getAlternatives) {
+          getAlternativesAsync(getAlternatives).then((entities) => {
+            if (initialize.current && (uidA === '' || uidB === '')) {
+              setFirstComparisonParams(
+                genInitialComparisonParams(entities, comparisons, uidA, uidB)
+              );
+            }
+
+            // stop loading after the initial comparison params have been built
+            setIsLoading(false);
+          });
+        } else {
+          // stop loading if no alternatives have been provided
+          setIsLoading(false);
+        }
+      });
+    } else {
+      // stop loading if no series is going to be rendered
+      setIsLoading(false);
+    }
```

Comment:
One way to make that a bit simpler maybe (and `Promise.all()` makes the 2 requests concurrent instead of sequential).

Feel free to apply it or not.

```suggestion
    if (length >= MIN_LENGTH) {
      const comparisonsPromise = getUserComparisonsAsync(pollName);
      const alternativesPromise = getAlternatives
        ? getAlternativesAsync(getAlternatives)
        : Promise.resolve();
      Promise.all([comparisonsPromise, alternativesPromise])
        .then(([comparisons, entities]) => {
          if (entities && initialize.current && (uidA === '' || uidB === '')) {
            setFirstComparisonParams(
              genInitialComparisonParams(entities, comparisons, uidA, uidB)
            );
          }
        })
        .then(() => {
          setIsLoading(false);
        });
    } else {
      // stop loading if no series is going to be rendered
      setIsLoading(false);
    }
```